hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileReader.java

```diff
@@ -343,16 +344,18 @@ private static GenericRecord deserialize(final byte[] keyBytes,
                                            final byte[] valueBytes,
                                            Schema writerSchema,
                                            Schema readerSchema) throws IOException {
-    GenericRecord record = HoodieAvroUtils.bytesToAvro(valueBytes, writerSchema, readerSchema);
+    GenericRecord wRecord = HoodieAvroUtils.bytesToAvro(valueBytes, writerSchema);
+    GenericData.Record rRecord = new GenericData.Record(readerSchema);
+    rRecord.getSchema().getFields().forEach(field -> rRecord.put(field.name(), wRecord.get(field.name())));
```

Comment:
reconstructing the record with new schema might incur some extra overhead. I think any field that gets added to schema should be nullable and/or have default.